src/cts/src/Util.h

```diff
@@ -135,9 +162,25 @@ class Box
   T getWidth() const { return width_; }
   T getHeight() const { return height_; }
 
-  Point<T> computeCenter() const
+  Point<T> getCenter()
+  {
+    if (centerSet_) {
+      Point<T> center(centerX_, centerY_);
+      return center;
+    }
+
+    centerX_ = xMin_ + width_ / 2;
+    centerY_ = yMin_ + height_ / 2;
+    Point<T> center(centerX_, centerY_);
+    centerSet_ = true;
+    return center;
+  }
+
+  void setCenter(Point<T> point)
```

Comment:
Why is a setter needed here?  Can the center ever be other than what getCenter would compute?

Reply:
Yes.  Center may get moved during legalization.

Reply:
So the center of the Box may not be what would ordinarily be considered the center?  That seems quite confusing.

Reply:
Correct, getCenter() may not be a geometric center after legalization.   What can I do to make this clearer?